### PR TITLE
Fix parsing from number

### PR DESCRIPTION
**Pull request by jcortes**
### <samp>🤖 Generated by Copilot at eb9cf2d</samp>

Use the `phone` library to validate and parse the `from` parameter for all Twilio actions. This improves consistency and error handling for the Twilio component.

### CODE REVIEW

File: components/twilio/actions/list-messages/list-messages.mjs [hunk 1]
```diff
@@ -47,9 +47,18 @@ export default {
       to = toParsed.phoneNumber;
     }
 
+    let from = this.from;
+    if (this.from) {
+      const fromParsed = phone(this.from);
+      if (!fromParsed || !fromParsed.phoneNumber) {
+        throw new Error(`Phone number ${this.from} couldn't be parsed as a valid number.`);
```
Comment (dylburger) [inline components/twilio/actions/list-messages/list-messages.mjs:54]:
```suggestion
        throw new Error(`Phone number ${this.from} could not be parsed as a valid number.`);
```

File: components/twilio/actions/make-phone-call/make-phone-call.mjs [hunk 2]
```diff
@@ -39,9 +39,14 @@ export default {
       throw new Error(`Phone number ${this.to} couldn't be parsed as a valid number.`);
     }
 
+    const fromParsed = phone(this.from);
+    if (!fromParsed || !fromParsed.phoneNumber) {
+      throw new Error(`Phone number ${this.from} couldn't be parsed as a valid number.`);
```
Comment (dylburger) [inline components/twilio/actions/make-phone-call/make-phone-call.mjs:44]:
```suggestion
      throw new Error(`Phone number ${this.from} could not be parsed as a valid number.`);
```

File: components/twilio/actions/send-mms/send-mms.mjs [hunk 3]
```diff
@@ -45,9 +45,14 @@ export default {
       throw new Error(`Phone number ${this.to} couldn't be parsed as a valid number.`);
     }
 
+    const fromParsed = phone(this.from);
+    if (!fromParsed || !fromParsed.phoneNumber) {
+      throw new Error(`Phone number ${this.from} couldn't be parsed as a valid number.`);
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/twilio/actions/send-mms/send-mms.mjs:50]: Changed 'couldn't' to 'could not' to follow Google's Developer docs style guide

File: components/twilio/actions/send-sms/send-sms.mjs [hunk 4]
```diff
@@ -39,9 +39,14 @@ export default {
       throw new Error(`Phone number ${this.to} couldn't be parsed as a valid number.`);
     }
 
+    const fromParsed = phone(this.from);
+    if (!fromParsed || !fromParsed.phoneNumber) {
+      throw new Error(`Phone number ${this.from} couldn't be parsed as a valid number.`);
```
Comment (dylburger) [inline components/twilio/actions/send-sms/send-sms.mjs:44]:
```suggestion
      throw new Error(`Phone number ${this.from} could not be parsed as a valid number.`);
```

Comment (dylburger) [reply components/twilio/actions/send-sms/send-sms.mjs:44]: Changed 'couldn't' to 'could not' to follow Google's Developer docs style guide

File: components/twilio/actions/list-messages/list-messages.mjs [hunk 5]
```diff
@@ -42,14 +42,23 @@ export default {
     if (this.to) {
       const toParsed = phone(this.to);
       if (!toParsed || !toParsed.phoneNumber) {
-        throw new Error(`Phone number ${this.to} couldn't be parsed as a valid number.`);
+        throw new Error(`Phone number ${this.to} could not be parsed as a valid number.`);
```
Comment (dylburger) [inline components/twilio/actions/list-messages/list-messages.mjs:45]:
```suggestion
        throw new Error(`Phone number ${this.to} couldn't be parsed as a valid number.`);
```

Comment (dylburger) [reply components/twilio/actions/list-messages/list-messages.mjs:45]: Changed 'could not' to 'couldn't' for conciseness (Google's Developer docs style guide)

File: components/twilio/actions/list-messages/list-messages.mjs [hunk 6]
```diff
@@ -42,14 +42,23 @@ export default {
...
       }
       to = toParsed.phoneNumber;
     }
 
+    let from = this.from;
+    if (this.from) {
+      const fromParsed = phone(this.from);
+      if (!fromParsed || !fromParsed.phoneNumber) {
+        throw new Error(`Phone number ${this.from} could not be parsed as a valid number.`);
```
Comment (dylburger) [inline components/twilio/actions/list-messages/list-messages.mjs:54]:
```suggestion
        throw new Error(`Phone number ${this.from} couldn't be parsed as a valid number.`);
```

Comment (dylburger) [reply components/twilio/actions/list-messages/list-messages.mjs:54]: Changed 'could not' to 'couldn't' for conciseness (Google's Developer docs style guide)

File: components/twilio/actions/make-phone-call/make-phone-call.mjs [hunk 7]
```diff
@@ -36,12 +36,17 @@ export default {
     const toParsed = phone(this.to);
     console.log(toParsed);
     if (!toParsed || !toParsed.phoneNumber) {
-      throw new Error(`Phone number ${this.to} couldn't be parsed as a valid number.`);
+      throw new Error(`Phone number ${this.to} could not be parsed as a valid number.`);
```
Comment (dylburger) [inline components/twilio/actions/make-phone-call/make-phone-call.mjs:39]:
```suggestion
      throw new Error(`Phone number ${this.to} couldn't be parsed as a valid number.`);
```

File: components/twilio/actions/make-phone-call/make-phone-call.mjs [hunk 8]
```diff
@@ -36,12 +36,17 @@ export default {
...
+    }
+
+    const fromParsed = phone(this.from);
+    if (!fromParsed || !fromParsed.phoneNumber) {
+      throw new Error(`Phone number ${this.from} could not be parsed as a valid number.`);
```
Comment (dylburger) [inline components/twilio/actions/make-phone-call/make-phone-call.mjs:44]:
```suggestion
      throw new Error(`Phone number ${this.from} couldn't be parsed as a valid number.`);
```

Comment (dylburger) [reply components/twilio/actions/make-phone-call/make-phone-call.mjs:44]: Changed 'could not' to 'couldn't' for conciseness (Google's Developer docs style guide)

File: components/twilio/actions/send-mms/send-mms.mjs [hunk 9]
```diff
@@ -42,12 +42,17 @@ export default {
     // See https://www.npmjs.com/package/phone
     const toParsed = phone(this.to);
     if (!toParsed || !toParsed.phoneNumber) {
-      throw new Error(`Phone number ${this.to} couldn't be parsed as a valid number.`);
+      throw new Error(`Phone number ${this.to} could not be parsed as a valid number.`);
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/twilio/actions/send-mms/send-mms.mjs:45]: Changed 'could not' to 'couldn't' for conciseness (Google's Developer docs style guide)

File: components/twilio/actions/send-mms/send-mms.mjs [hunk 10]
```diff
@@ -42,12 +42,17 @@ export default {
...
+    }
+
+    const fromParsed = phone(this.from);
+    if (!fromParsed || !fromParsed.phoneNumber) {
+      throw new Error(`Phone number ${this.from} could not be parsed as a valid number.`);
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/twilio/actions/send-mms/send-mms.mjs:50]: Changed 'could not' to 'couldn't' for conciseness (Google's Developer docs style guide)

File: components/twilio/actions/send-sms/send-sms.mjs [hunk 11]
```diff
@@ -36,12 +36,17 @@ export default {
     // See https://www.npmjs.com/package/phone
     const toParsed = phone(this.to);
     if (!toParsed || !toParsed.phoneNumber) {
-      throw new Error(`Phone number ${this.to} couldn't be parsed as a valid number.`);
+      throw new Error(`Phone number ${this.to} could not be parsed as a valid number.`);
```
Comment (dylburger) [inline components/twilio/actions/send-sms/send-sms.mjs:39]:
```suggestion
      throw new Error(`Phone number ${this.to} couldn't be parsed as a valid number.`);
```

Comment (dylburger) [reply components/twilio/actions/send-sms/send-sms.mjs:39]: Changed 'could not' to 'couldn't' for conciseness (Google's Developer docs style guide)

File: components/twilio/actions/send-sms/send-sms.mjs [hunk 12]
```diff
@@ -36,12 +36,17 @@ export default {
...
+    }
+
+    const fromParsed = phone(this.from);
+    if (!fromParsed || !fromParsed.phoneNumber) {
+      throw new Error(`Phone number ${this.from} could not be parsed as a valid number.`);
```
[the remark this answers is not in the document]
Comment (dylburger) [reply components/twilio/actions/send-sms/send-sms.mjs:44]: Changed 'could not' to 'couldn't' for conciseness (Google's Developer docs style guide)

**Review by alysonturing — APPROVED** [review]
LGTM!

**Comment by vunguyenhung** [comment]
As the code is mostly duplicated from other prop, and since I don't have the supported phone number to send sms or msm to, I will move this to ready for release and track for any issue.

**Comment by gustavorps** [comment]
Error
Phone number +552712345678 could not be parsed as a valid number.

```
DETAILS
    at Object.run (file:///tmp/__pdg__/dist/code/d1e867980c79ca35395bfc545bb5f8daa49e98ade92927565d22a0e32f3a00f1/code/actions/make-phone-call/make-phone-call.mjs:44:13)
    at null.executeComponent (/var/task/launch_worker.js:229:53)
    at MessagePort.messageHandler (/var/task/launch_worker.js:726:28)
```

+552712345678 is a valid number in my country (Brazil)

**Pull request merged**